app/src/main/java/io/left/meshim/controllers/RightMeshController.java

```diff
@@ -285,7 +296,12 @@ private void handlePeerChanged(RightMeshEvent e) {
             byte[] message = createPeerUpdatePayloadFromUser(user);
             try {
                 if (message != null) {
-                    meshManager.sendDataReliable(event.peerUuid, MESH_PORT, message);
+                   int dataId = meshManager.sendDataReliable(event.peerUuid, MESH_PORT, message);
+                   //making the status of the message undelivered so we can resend the package if delivery
+                    // report is not recieved.
+                   undeliveredPeerUpdateMessage.put(dataId,event.peerUuid);
+                   // run a timer for undelivered peer updates.
```

Comment:
Something like "schedule a task to resend undelivered peer updates" might be more clear than "run a timer".